rumcake-macros/src/keyboard.rs

```diff
@@ -345,6 +346,22 @@ pub(crate) fn keyboard_main(
         ::rumcake::hw::mcu::initialize_rcc();
     });
 
+    if let Some(arg) = keyboard.bootloader_double_tap_reset {
+        let timeout = arg.unwrap_or(200);
+
+        if timeout == 0 {
+            initialization.extend(quote! {
+                compile_error!("The timeout for double tapping the reset button should be > 0");
+            })
+        }
+
+        initialization.extend(quote! {
+            unsafe {
+                ::rumcake::hw::check_double_tap_bootloader(#timeout).await;
+            }
+        });
+    }
+
```

Comment:
Realized in testing that this should be moved to the end of the macro, otherwise it will prevent the other tasks from spawning until the end of the specified timeout. So basically just move this block to the bottom of the function (below display setup), and write the tokens to `spawning` instead of `initialization`.